### PR TITLE
feat(backend): aws ses for mails tm-611

### PR DESCRIPTION
**Pull request by Ann-Maksay**
image

### CODE REVIEW

File: apps/backend/src/libs/modules/mail/mail.module.ts [hunk 1]
```diff
@@ -62,10 +45,27 @@ class Mail {
...
+					},
+				},
+				Subject: {
+					Data: subject,
+				},
+			},
+			Source: this.sender,
+		};
+
+		const sendEmainCommand = new SendEmailCommand(parameters);
```
Comment (fshabanov) [inline apps/backend/src/libs/modules/mail/mail.module.ts:65]:
```suggestion
		const sendEmailCommand = new SendEmailCommand(parameters);
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/.env.example [hunk 2]
```diff
@@ -22,12 +22,13 @@ TOKEN_ALGORITHM=HS256
 TOKEN_EXPIRES_IN=24h
 
 #
-# AWS S3
+# AWS
 #
 AWS_ACCESS_KEY_ID=aws_access_key_id
 AWS_SECRET_ACCESS_KEY=aws_secret_access_key
 S3_REGION=s3_region
```
Comment (what1s1ove) [inline apps/backend/.env.example:29]:
```suggestion
AWS_REGION=s3_region
```
you now use it more that just for s3

File: apps/backend/.env.example [hunk 3]
```diff
@@ -22,12 +22,13 @@ TOKEN_ALGORITHM=HS256
 TOKEN_EXPIRES_IN=24h
 
 #
-# AWS S3
+# AWS
 #
 AWS_ACCESS_KEY_ID=aws_access_key_id
 AWS_SECRET_ACCESS_KEY=aws_secret_access_key
```
Comment (what1s1ove) [inline apps/backend/.env.example:28]:
```suggestion
AWS_ACCESS_KEY=aws_access_key
AWS_SECRET_KEY=aws_secret_key
```
let's simplify names

File: apps/backend/src/libs/modules/config/base-config.module.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ class BaseConfig implements Config {
 					format: Number,
 				},
 			},
-			AWS_S3: {
+			AWS: {
 				AWS_ACCESS_KEY_ID: {
```
Comment (what1s1ove) [inline apps/backend/src/libs/modules/config/base-config.module.ts:54]:
```suggestion
				ACCESS_KEY: {
```
and others
ACCESS_KEY
REGION

File: apps/backend/src/libs/modules/mail/mail.module.ts [hunk 5]
```diff
@@ -7,49 +7,32 @@ import { MailError } from "./libs/exceptions/exceptions.js";
...
 type Constructor = {
 	config: {
-		isLogged: boolean;
-		isRequireTLS: boolean;
-		isSecure: boolean;
-		senderEmail: string;
-		senderName: string;
-		senderPassword: string;
-		service: string;
+		accessKeyId: string;
```
Comment (what1s1ove) [inline apps/backend/src/libs/modules/mail/mail.module.ts:10]:
```suggestion
		accessKey: string;
```

File: apps/backend/src/libs/modules/mail/mail.module.ts [hunk 6]
```diff
@@ -7,49 +7,32 @@ import { MailError } from "./libs/exceptions/exceptions.js";
...
+		region: string;
+		secretAccessKey: string;
```
Comment (what1s1ove) [inline apps/backend/src/libs/modules/mail/mail.module.ts:12]:
```suggestion
		secretKey: string;
```

File: apps/backend/src/modules/files/files.ts [hunk 7]
```diff
@@ -10,10 +10,10 @@ import { FileService } from "./file.service.js";
...
 const fileService = new FileService({
 	credentials: {
-		accessKeyId: config.ENV.AWS_S3.AWS_ACCESS_KEY_ID,
-		bucket: config.ENV.AWS_S3.S3_BUCKET,
-		region: config.ENV.AWS_S3.S3_REGION,
-		secretAccessKey: config.ENV.AWS_S3.AWS_SECRET_ACCESS_KEY,
+		accessKeyId: config.ENV.AWS.AWS_ACCESS_KEY_ID,
+		bucket: config.ENV.AWS.S3_BUCKET,
+		region: config.ENV.AWS.S3_REGION,
+		secretAccessKey: config.ENV.AWS.AWS_SECRET_ACCESS_KEY,
```
Comment (what1s1ove) [inline apps/backend/src/modules/files/files.ts:16]:
```suggestion
		accessKey: config.ENV.AWS.AWS_ACCESS_KEY_ID,
		bucket: config.ENV.AWS.S3_BUCKET,
		region: config.ENV.AWS.S3_REGION,
		secretKey: config.ENV.AWS.AWS_SECRET_ACCESS_KEY,
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/modules/mail/mail.module.ts (hunk 5 above)
Comment (Ann-Maksay) [reply apps/backend/src/libs/modules/mail/mail.module.ts:10]: AWS clients expect these key names for credentials, I'll change them here just to write the client call like this:
```
this.sesClient = new SESClient({
	credentials: {
		accessKeyId: accessKey,
		secretAccessKey: secretKey,
	},
	region,
});
```
Is that really worth doing? @what1s1ove

File: apps/backend/src/libs/modules/mail/mail.module.ts (hunk 6 above)
Comment (Ann-Maksay) [reply apps/backend/src/libs/modules/mail/mail.module.ts:12]: Same [here](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-trackmates/pull/641#discussion_r1532250270l)

File: apps/backend/src/modules/files/files.ts (hunk 7 above)
Comment (Ann-Maksay) [reply apps/backend/src/modules/files/files.ts:16]: Same [here](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-trackmates/pull/641#discussion_r1532250270l)

File: apps/backend/src/libs/modules/mail/mail.module.ts (hunk 5 above)
Comment (what1s1ove) [reply apps/backend/src/libs/modules/mail/mail.module.ts:10]: yes, do it

File: apps/backend/src/libs/modules/mail/mail.module.ts (hunk 6 above)
Comment (what1s1ove) [reply apps/backend/src/libs/modules/mail/mail.module.ts:12]: do it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/modules/mail/mail.module.ts [hunk 8]
```diff
@@ -62,10 +45,27 @@ class Mail {
 		subject: string;
 		text: string;
 	}): Promise<boolean> {
-		const from = `${this.senderName} <${this.senderEmail}>`;
+		const parameters = {
```
Comment (fshabanov) [inline apps/backend/src/libs/modules/mail/mail.module.ts:48]: No need to create a separate variable for that

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/modules/mail/mail.module.ts [hunk 9]
```diff
@@ -62,10 +45,25 @@ class Mail {
 		subject: string;
 		text: string;
 	}): Promise<boolean> {
-		const from = `${this.senderName} <${this.senderEmail}>`;
+		const sendEmailCommand = new SendEmailCommand({
+			Destination: {
+				ToAddresses: [to],
```
Comment (KrasKa2) [inline apps/backend/src/libs/modules/mail/mail.module.ts:50]:
```suggestion
				ToAddresses: [email],
```
and remove renaming `email: to`

**Review by KrasKa2 — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**